File: e24.erl
-module(e24).
-on_load(e24/0).

fac(0) -> 1;
fac(N) -> N * fac(N - 1).


list_ref(0, [Head|_]) -> Head;
list_ref(X, [_|Tail]) -> list_ref(X - 1, Tail).

nthLP(N, P, Rem) ->
    if
        length(Rem) == 1 ->
            P ++ Rem;
        true ->
            K    = fac(9 - length(P)),
            Elem = list_ref(trunc(N / K), Rem),
            
            nthLP(N rem  K, P ++ [Elem], Rem -- [Elem])
    end.
e24() -> io:format("~B~B~B~B~B~B~B~B~B~B~n", 
    nthLP(999999, [], [0,1,2,3,4,5,6,7,8,9])).

File: e24.py
def fac(x):
    return 1 if x < 2 else x * fac(x-1)

#p is the list of predetermined digits (in order)
#rem is the list of digits left
def NthLP(N, p, rem):
    if len(rem) == 1:
        return p + rem
    k = fac(len(rem) - 1)
    timesDivided = int(N / k)
    print(timesDivided +1)
    p.append(rem[timesDivided])
    rem.remove(p[-1])
    return NthLP(N % k, p, rem)

digits = NthLP(int(999999), [], [i for i in range(0,10)])
s = ""
for d in digits:
    s += str(d)
print s
